File: DSA.BLL/Services/Facilities/DepartmentService.cs
using RCS.BLL.Dto.Facilities;
using RCS.BLL.Services.Contracts;
using RCS.DAL.UnitOfWork.Contract;
using RCS.Domain.Facilities;
using RCS.Domain.Params;
using System.Collections.Generic;

namespace RCS.BLL.Services
{
    public class DepartmentService : IDepartmentService
    {
        private readonly IUnitOfWork _unitOfWork;

        public DepartmentService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }
        
        public void AddDepartment(AddDepartmentDto data)
        {
            var newDepartment = AutoMapper.Mapper.Map<AddDepartmentDto, Department>(data);
            _unitOfWork.DepartmentRepository.Add(newDepartment);
            _unitOfWork.Commit();
        }
        
        public CollectionResult<DepartmentDto> GetDepartmentsByParams(DepartmentsFilterParams filterParams)
        {
            var items = _unitOfWork.DepartmentRepository.GetDepartmentsByParams(filterParams);

            var result = new CollectionResult<DepartmentDto>
            {
                TotalCount = items.TotalCount,
                Collection = AutoMapper.Mapper.Map<IEnumerable<Department>, List<DepartmentDto>>(items.Collection)
            };

            return result;
        }

        public IEnumerable<DepartmentDto> GetDepartmentsByTerm(int facilityId, string term)
        {
            var items = _unitOfWork.DepartmentRepository.GetDepartmentsByTerm(facilityId, term);

            return AutoMapper.Mapper.Map<IEnumerable<DepartmentDto>>(items);
        }
    }
}

File: DSA.SPA/ClientApp/pages/facility/api/department-service.js
import * as httpService from '../../../api/http-service';

const apiDepartment = "/api/department/";

export const getDepartmentsByTerm = (term) => {
    let params = {
        url: apiDepartment + `GetDepartmentsByTerm?term=${term}`
    }

    return httpService.getData(params);
}

export const getDepartments = data => {
    let params = {
        url: apiDepartment + "GetDepartments",
        data: data
    }

    return httpService.postData(params);
}

export const addDepartment = data => {
    let params = {
        url: apiDepartment + "AddDepartment",
        data: data
    }

    return httpService.postData(params);
}